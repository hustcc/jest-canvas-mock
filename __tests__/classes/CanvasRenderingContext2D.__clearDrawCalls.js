let ctx;
beforeEach(() => {
  // get a new context each test
  ctx = document.createElement('canvas').getContext('2d');
});

afterEach(() => {
  const drawCalls = ctx.__getDrawCalls();
  expect(drawCalls).toMatchSnapshot();
});

describe('__clearDrawCalls', () => {
  it('should clear the list of draw calls', () => {
    ctx.fillRect(1, 2, 3, 4);
    ctx.__clearDrawCalls();
  });

  it('should not prevent additional draw calls from being collected', () => {
    ctx.fillRect(1, 2, 3, 4);
    ctx.__clearDrawCalls();
    ctx.fillRect(1, 2, 3, 4);
  });
});
