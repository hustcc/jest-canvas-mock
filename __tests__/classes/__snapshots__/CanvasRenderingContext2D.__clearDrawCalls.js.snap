// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`__clearDrawCalls should clear the list of draw calls 1`] = `Array []`;

exports[`__clearDrawCalls should not prevent additional draw calls from being collected 1`] = `
Array [
  Object {
    "props": Object {
      "height": 4,
      "width": 3,
      "x": 1,
      "y": 2,
    },
    "type": "fillRect",
  },
]
`;
